diff --git a/node_modules/@sap/cds/server.js b/node_modules/@sap/cds/server.js
index 5bbf08c..17f5506 100644
--- a/node_modules/@sap/cds/server.js
+++ b/node_modules/@sap/cds/server.js
@@ -1,6 +1,6 @@
 const express = require('express')// eslint-disable-line cds/no-missing-dependencies
 const cds = require('./lib')
-
+const { Worker, isMainThread } = require("node:worker_threads")
 /**
  * Standard express.js bootstrapping, constructing an express `application`
  * and launching a corresponding http server using `app.listen()`.
@@ -41,8 +41,10 @@ module.exports = async function cds_server (options) {
   await cds.emit ('served', cds.services)
 
   // start http server
-  const port = o.port !== undefined ? o.port : ( process.env.PORT || cds.env.server?.port || 4004 )
-  return app.server = app.listen (port)
+  if (isMainThread) {
+    const port = o.port !== undefined ? o.port : ( process.env.PORT || cds.env.server?.port || 4004 )
+    return app.server = app.listen (port)
+  }
 
   // cds.deploy in-memory db, if enabled
   async function _init (db) {
